policies/entity_collisions.rego

```diff
@@ -0,0 +1,48 @@
+package after_resolution
+
+import rego.v1
+
+# Data structures to make checking things faster.
+entities_names := { obj |
+    group := input.groups[_]
+    group.type = "entity"
+    obj := { "name": group.name, "namespace_prefix": extract_entity_namespace_prefix(group.name), "deprecated": is_property_set(group, "deprecated")}
+}
+
+# check that entity names do not collide with namespaces
+deny contains entities_registry_collision(description, name) if {
+    some i
+
+    name := entities_names[i].name
+    prefix := entities_names[i].namespace_prefix
+    not entities_names[i].deprecated
+
+    collisions := [other.name |
+        other := entities_names[_]
+        not other.deprecated
+
+        other.name != name
+        startswith(other.name, prefix)
+    ]
+    count(collisions) > 0
+    description := sprintf("Entity with name '%s' is used as a namespace in the following entities '%s'.", [name, collisions])
+}
+
+
+entities_registry_collision(description, entity_name) = violation if {
```

Comment:
So `entity_type` is *intended* to be a namespace for entities.

The is for an entity type `k8s.pod`, we expect attributes to be `k8s.pod.*` generally.  We aren't making this a hard requirement yet, but that's the intention.

I realized, that should be updated in the [modelling guide](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/non-normative/how-to-write-conventions/resource-and-entities.md#how-to-namespace-entities).